### PR TITLE
enable PodSecurityPolicy

### PR DESCRIPTION
**Pull request by szuecs**
This can be used for a discussion on what kind of PSP we want to have.

This PR shows how to enable PSP and use a hack in the webhook to disallow PowerUsers to create privileged containers. The webhook will only allow restricted PODs for PowerUsers.

### CODE REVIEW

File: cluster/manifests/psp/pod_security_policy.yaml [hunk 1]
```diff
@@ -33,4 +33,10 @@ spec:
   supplementalGroups:
     rule: RunAsAny
   volumes:
-  - '*'
+  - emptyDir
+  - awsElasticBlockStore
```
Comment (hjacobs) [inline cluster/manifests/psp/pod_security_policy.yaml]: do we need `awsElasticBlockStore`? isn't this working via `persistentVolumeClaim`?

Comment (szuecs) [inline cluster/manifests/psp/pod_security_policy.yaml]: I don't know, but I also don't think we should disable it, if someone has a reason to use it.

**Comment by szuecs** [comment]
@hjacobs @mikkeloscar @ideahitme @Raffo @tuxlife @linki please review now, in my opinion it's good to go

File: cluster/cluster.py [hunk 2]
```diff
@@ -396,8 +396,12 @@ def longest_grace_period(node_name: str, config: dict):
     """
     headers = {"Authorization": "Bearer {}".format(config["worker_shared_secret"])}
     params = {"fieldSelector": "spec.nodeName={}".format(node_name)}
-    pods = requests.get(config["api_server"] + "/api/v1/pods", params=params, headers=headers, timeout=5).json()
+    resp = requests.get(config["api_server"] + "/api/v1/pods", params=params, headers=headers, timeout=5)
+    pods = resp.json()
     grace_period = 0
+    if not pods:
+        print("resp:", resp)
```
Comment (Raffo) [inline cluster/cluster.py]: Is this a debugging leftover? If not can you make the print more informative?

File: cluster/userdata-master.yaml [hunk 3]
```diff
@@ -354,11 +354,11 @@ write_files:
           resources:
             requests:
               cpu: 100m
-              memory: 500Mi
+              memory: 200Mi
             limits:
               cpu: 200m
               memory: 1Gi
-        - image: registry.opensource.zalan.do/teapot/k8s-authnz-webhook:v0.1.0
+        - image: registry.opensource.zalan.do/teapot/k8s-authnz-webhook:v0.1.0-6-g372fc9f
```
Comment (Raffo) [inline cluster/userdata-master.yaml]: Can we tag the webhook and make sure to use the right image? (i.e. v0.1.1)

Comment (szuecs) [inline cluster/userdata-master.yaml]: done

File: cluster/cluster.py (hunk 2 above)
Comment (szuecs) [inline cluster/cluster.py]: done

**Comment by Raffo** [comment]
👍

**Comment by szuecs** [comment]
:+1:

**Pull request merged**